src/tests/unit/auto/exec_network_get_metrics.cpp

```diff
@@ -173,7 +122,117 @@ class ExecNetworkGetMetric : public ::testing::TestWithParam<ConfigParams> {
+                                int,           // cpu infer requet num of customer want
+                                bool,          // if cpu sleep, cpu device will load slow
+                                unsigned int,  // Actual device OPTIMAL_NUMBER_OF_INFER_REQUESTS
+                                int,           // Actual device infer requet num of customer want
+                                bool,          // if Actual device sleep, cpu device will load slow
+                                std::string,   // Actual Device Name
+                                unsigned int,  // expect OPTIMAL_NUMBER_OF_INFER_REQUESTS
+                                int            // Actual PERFORMANCE_HINT_NUM_REQUESTS
+                                >;
+class ExecNetworkGetMetricOptimalNumInferReq : public ExecNetworkGetMetricBase,
+                                               public ::testing::WithParamInterface<ConfigParams> {
+public:
+    static std::string getTestCaseName(testing::TestParamInfo<ConfigParams> obj) {
+        unsigned int cpuOptimalNum;
+        int cpuCustomerNum;
+        unsigned int actualOptimalNum;
+        int actualCustomerNum;
+        unsigned int expectOptimalNum;
+        bool cpuSleep;
+        bool actualSleep;
+        bool isThroughput;
+        int gpuPerfHintNum;
+        std::string actualDeviceName;
+        std::tie(isThroughput,
+                 cpuOptimalNum,
+                 cpuCustomerNum,
+                 cpuSleep,
+                 actualOptimalNum,
+                 actualCustomerNum,
+                 actualSleep,
+                 actualDeviceName,
+                 expectOptimalNum,
+                 gpuPerfHintNum) = obj.param;
+        std::ostringstream result;
+        result << "cpuOptimalNum_" << cpuOptimalNum << "cpuCustomerNum_" << cpuCustomerNum;
+        result << "actualOptimalNum_" << actualOptimalNum << "actualCustomerNum_" << actualCustomerNum;
+        result << "expectOptimalNum_" << expectOptimalNum;
+        if (isThroughput) {
+            result << "_isThroughput"
+                   << "true";
+        } else {
+            result << "__isThroughput"
+                   << "false";
+        }
+        if (cpuSleep) {
+            result << "_cpuSleep_"
+                   << "true";
+        } else {
+            result << "_cpuSleep_"
+                   << "false";
+        }
+
+        if (actualSleep) {
+            result << "_actualSleep_"
+                   << "true";
+        } else {
+            result << "_actualSleep_"
+                   << "false";
+        }
+        result << "_actualDeviceName_" << actualDeviceName;
+        result << "_gpuPerfHintNum_" << gpuPerfHintNum;
+        return result.str();
+    }
+};
+
+using ConfigParams1 = std::tuple<bool,          // is New API
+                                 bool,          // if Actual device sleep, cpu device will load slow
```

Comment:
better use a meaningful name instead of ConfigParams1

Reply:
Updated